airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -249,25 +279,139 @@ async def test_state_with_abnormally_large_values(
-        if is_per_stream:
-            # Including all the latest state values from previous batches, update the combined stream state
-            # with the current batch's stream state and then use it in the following read() request
-            current_state = state_message.state
-            if current_state and current_state.type == AirbyteStateType.STREAM:
-                per_stream = current_state.stream
-                if per_stream.stream_state:
-                    stream_name_to_per_stream_state[per_stream.stream_descriptor.name] = (
-                        per_stream.stream_state.dict() if per_stream.stream_state else {}
-                    )
-            state_input = [
-                {"type": "STREAM", "stream": {"stream_descriptor": {"name": stream_name}, "stream_state": stream_state}}
-                for stream_name, stream_state in stream_name_to_per_stream_state.items()
-            ]
-            return state_input, stream_name_to_per_stream_state
-        else:
-            return state_message.state.data, state_message.state.data
+        # Including all the latest state values from previous batches, update the combined stream state
+        # with the current batch's stream state and then use it in the following read() request
+        current_state = state_message.state
+        if current_state and current_state.type == AirbyteStateType.STREAM:
+            per_stream = current_state.stream
+            if per_stream.stream_state:
+                stream_name_to_per_stream_state[per_stream.stream_descriptor.name] = (
+                    per_stream.stream_state.dict() if per_stream.stream_state else {}
+                )
+        state_input = [
+            {"type": "STREAM", "stream": {"stream_descriptor": {"name": stream_name}, "stream_state": stream_state}}
+            for stream_name, stream_state in stream_name_to_per_stream_state.items()
+        ]
+        return state_input, stream_name_to_per_stream_state
+
+    def get_unique_state_messages(self, states: List[AirbyteStateMessage]) -> List[AirbyteStateMessage]:
+        """
+        Validates a list of state messages to ensure that consecutive messages with the same stream state are represented by only the first message, while subsequent duplicates are ignored.
+        """
+        # If there is only one state message or less, the list is considered to be unique
+        if len(states) <= 1:
+            return states
+
+        stream_state_attr = "state.stream.stream_state"
+        record_count_attr = "state.sourceStats.recordCount"
+
+        # Define a function to get the stream state attribute from an AirbyteStateMessage object
+        get_state = operator.attrgetter(stream_state_attr)
+
+        # Define a function to get the record count attribute from an AirbyteStateMessage object
+        get_record_count = operator.attrgetter(record_count_attr)
+
+        current_idx = 0
+        result = []
```

Comment:
let's rename this `result` variable since it doesn't represent the result of the method. It seems to represent the list of unique state messages

Reply:
Done